src/core/meshcop/border_agent.hpp

```diff
@@ -330,6 +331,7 @@ class BorderAgent : public InstanceLocator, private NonCopyable
     Ip6::Udp::Receiver         mUdpReceiver;
     Ip6::Netif::UnicastAddress mCommissionerAloc;
     TimeoutTimer               mTimer;
+    bool                       mPskcChanged;
```

Comment:
Just a suggestion to move this up after `mState` (so the member variable fields may be packed better?)